toolkits/landcover/impl/named-args.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Regular expression used to extract argument names from string represention
+ * of a function.
+ */
+var FUNCTION_REGEX = /^function\s*(?:.*?)\(\s*(.*)\s*\)/;
```

Comment:
This regex doesn't look right.  The args section needs to exclude ')' or something like "function (a,b,c) { d(e,f); }" will match all the way to the end of the d function inside the body.  Also, since you're stripping whitespace anyway, you can simplify and ignore the spaces inside the (), and you're not using the function name, so you don't need to capture it.

So I think you end up with this: /^function\s[^(]*\\(([^)]\*)\\)/
Careful, github is hiding some characters here, and I had to extra escape them to make them visible.
This should read: "function", whitespace, any "not '('", "(", capture of: any not ")", ")".
I'd suggest documenting the regex contents for the future.

Also, it's possible to get comments in the args section.  ie: function a(b, /* comment */ c).  You should maybe strip them when you're stripping whitespace: arg.replace(/\/\/.*|\/\*[^]*?\*\//g, "")).trim()

Reply:
> This regex doesn't look right. The args section needs to exclude ')' or something like "function (a,b,c) { d(e,f); }" will match all the way to the end of the d function inside the body. 

Good catch. This was working because `.` doesn't match LFs in JavaScript by default; it would break if the args were on multiple lines. We can work around this by replacing all LFs with spaces first, which should be ok if we're not executing this code. We can then use non-greedy Kleene star to only match up to the first `)`.

>Also, since you're stripping whitespace anyway, you can simplify and ignore the spaces inside the (), 

Sure. The `\s` was there to capture new lines since `.` wasn't matching them, but that sol'n was incomplete. With the fix mentioned about we can removed them now.

>and you're not using the function name, so you don't need to capture it.

We're not capturing it (`(?:...)`), we're using it to ignore an optional function name if present. But this can be simplified with non greedy wildcard `.*?`

You can test this regexp on a real function decl here:
https://regex101.com/r/iAKuBM/1

> Also, it's possible to get comments in the args section. ie: function a(b, /* comment _/ c). You should maybe strip them when you're stripping whitespace: arg.replace(///._|/*[^]*?*//g, "")).trim()

Why would we ever add comments to the function decl? Isn't that what JSDoc is for?

Reply:
> Why would we ever add comments to the function decl? Isn't that what JSDoc is for?

We wouldn't, but other people might; we don't get to dictate style so we should be resilient. Thinking about this specific issue a little more, just stripping comments (and newlines) before any processing seems simplest.

Reply:
Since we fail fast here, implementors would quickly realize their function signature wasn't supported and would have a chance to fix it. Not sure how much value this adds, but implementing anyway so we can move forward.

Reply:
Actually this is simple enough to do, and since we don't fail fast on this (it's not handled in the main regex), going to implement it.